### ISSUE TITLE
[question] accesing option when exporting Draw/Impress to saving format

### ISSUE THREAD

**Issue by YouvenZ** [Amourspirit/python-ooouno-ex#287]
Hi @Amourspirit !

I would like to get acces to the option presented when saving a draw/Impress doc

![Screenshot from 2024-01-13 19-11-59](https://github.com/Amourspirit/python-ooouno-ex/assets/56958432/d5e729a7-5494-409f-8b46-44c52a8bc10b)
First I tried to change the size slide directly by changing the property height and weight
```
    def main(self) -> None:
        # connect headless. will not need to see slides
        with Lo.Loader(Lo.ConnectPipe(headless=True)) as loader:
            doc = Lo.open_doc(fnm=self._fnm, loader=loader)

            if not Info.is_doc_type(obj=doc, doc_type=Lo.Service.IMPRESS):
                Lo.print("-- Not a slides presentation")
                return

            slide = Draw.get_slide(doc=doc, idx=self._idx)
            slide.setPropertyValue("Width", 300)
            slide.setPropertyValue("Height", 200)
            ...
```
 but I was wrong it change actual size of the draw not the resolution. I want to change the resolution and the transparency of the image saved. could you provide some guidance on this please.

**Comment by Amourspirit** [comment]
I am not sure of the solution for this just yet. At this time `OooDev` does not have anything built in for this.

I am not even sure this is part of the LibreOffice API.

I did just today add an export Calc Range as Image. The demo is in this repo [here](https://github.com/Amourspirit/python-ooouno-ex/tree/main/ex/auto/calc/odev_export_calc_image).

The method to acomplish this uses filters. See my source for this [here](https://github.com/Amourspirit/python_ooo_dev_tools/blob/bc2a654b35c1353c2173a6bd4613f6a5f8cfef96/ooodev/calc/calc_cell_range.py#L508C9-L508C24).

I am saying this because I suspect that the implemention you are looking for is similar for Draw.

In `OooDev` See docs on [Info.get_filter_names()](https://python-ooo-dev-tools.readthedocs.io/en/latest/src/utils/info.html#ooodev.utils.info.Info.get_filter_names) and [Info.get_filter_props()](https://python-ooo-dev-tools.readthedocs.io/en/latest/src/utils/info.html#ooodev.utils.info.Info.get_filter_props) to get more information on filters.

See also https://ask.libreoffice.org/t/export-as-png-with-macro/74337/11
and https://ask.libreoffice.org/t/how-to-export-cell-range-to-images/57828/2

**Comment by YouvenZ** [comment]
Thanks for the answer! In the the example with Calc, the dict `event_data`  seems to have exactly what I need. I will take a look at the documentation to understand filters and try. I will also try to post the example if I find the correct workaround.

**Comment by Amourspirit** [comment]
I have been attempting to work out filters and add them to `OooDev`. So far it seems to be challenging and there may be bugs in Python around filters and Image export.

I starteded a post on Ask: [Basic works, Python fails same task. Exporting image fails for other resolutions. Why?](https://ask.libreoffice.org/t/basic-works-python-fails-same-task-exporting-image-fails-for-other-resolutions-why/100724)

**Comment by Amourspirit** [comment]
A new version of `OooDev` has been released to day, version `0.23.0`.

Now it is possible to export Draw Page and Impress Pages with filters.
This should give you what you need.

The [Impress Slide to Image](https://github.com/Amourspirit/python-ooouno-ex/tree/main/ex/auto/impress/odev_slide_to_image) example has been updated to reflect the usage of filters.

Also there is an [Impress Slide to Image 2](https://github.com/Amourspirit/python-ooouno-ex/tree/main/ex/auto/impress/odev_slide_to_image2) example that shows how export pages using filters in a different way.

Support has been added for exporting indivitual shapes a images too. The [Export Shape as Image](https://github.com/Amourspirit/python-ooouno-ex/tree/main/ex/auto/draw/odev_shape_export_img) demonstrates this.
